### PR TITLE
Support auth and unsignedPayload

### PR DESCRIPTION
**Pull request by mullermp**
Redo of https://github.com/aws/aws-sdk-ruby/pull/2923

Adds unsignedPayload and some extra testing.

### CODE REVIEW

File: build_tools/services.rb [hunk 1]
```diff
@@ -158,9 +158,16 @@ def gem_dependencies(api, dependencies)
       core_version_string = "', '>= #{min_core}"
       dependencies['aws-sdk-core'] = "~> #{version_file.split('.')[0]}#{core_version_string}"
 
+      api['metadata'].fetch('auth', []).each do |auth|
+        if %w[aws.auth#sigv4 aws.auth#sigv4a].include?(auth)
+          dependencies['aws-sigv4'] = '~> 1.5'
```
Comment (jterapin) [inline build_tools/services.rb:163]: Curious to what significant happened with aws-sigv4 1.5 for this pessimistic version constraint?

Comment (mullermp) [reply build_tools/services.rb:163]: CRT for sigv4a.

File: gems/aws-sdk-core/lib/aws-sdk-core/plugins/transfer_encoding.rb [hunk 2]
```diff
@@ -5,7 +5,8 @@ module Plugins
 
     # For Streaming Input Operations, when `requiresLength` is enabled
     # checking whether `Content-Length` header can be set,
-    # for `v4-unsigned-body` operations, set `Transfer-Encoding` header
+    # for `unsignedPayload` and v4-unsigned-body` operations,
```
Comment (jterapin) [inline gems/aws-sdk-core/lib/aws-sdk-core/plugins/transfer_encoding.rb]: Missing backtick in front of `v4-unsigned-body`

**Review by jterapin — APPROVED** [review]
🚢✨

**Pull request merged**